beater/beater.go

```diff
@@ -969,3 +976,45 @@ func (r *chanReloader) serve(ctx context.Context, reloader reload.Reloadable) er
 		}
 	}
 }
+
+// TODO: This is copying behavior from libbeat:
+// https://github.com/elastic/beats/blob/b9ced47dba8bb55faa3b2b834fd6529d3c4d0919/libbeat/cmd/instance/beat.go#L927-L950
+// Remove this when cluster_uuid no longer needs to be queried from ES.
+func queryClusterUUID(ctx context.Context, esClient elasticsearch.Client) error {
+	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
+	outputES := "outputs.elasticsearch"
+	elasticsearchRegistry := stateRegistry.GetRegistry(outputES)
+	if elasticsearchRegistry == nil {
+		elasticsearchRegistry = stateRegistry.NewRegistry(outputES)
+	}
+	// TODO: How can I just update a registry value?
+	clusterUUID := "cluster_uuid"
+	if elasticsearchRegistry.Get(clusterUUID) != nil {
+		elasticsearchRegistry.Remove(clusterUUID)
+	}
+	clusterUUIDRegVar := monitoring.NewString(elasticsearchRegistry, clusterUUID)
```

Comment:
I looked through the api but it wasn't immediately apparent to me how to update a value already existing in a registry.

Reply:
There's a `Get` method on the `monitoring.Registry` which I think you can use?

Would the libbeat code have been called yet at this point? I'm wondering though if there's any risk that the libbeat code could kick in concurrently and panic.

Reply:
>There's a Get method on the monitoring.Registry which I think you can use?

roger, I have to cast to `*monitoring.String`, but it appears to work (the return value is `Var` interface type, it seems, which only exposes a single `Visit` method).

>Would the libbeat code have been called yet at this point? I'm wondering though if there's any risk that the libbeat code could kick in concurrently and panic.

The registry code is called synchronously, before `*beater.Run`, so it should always be set up. The `Set` method on the `*monitoring.String` could be called at the same time, but it has an internal mutex, so access should be fine.

Reply:
Cool, thanks. I see you removed the test guards, looks much safer to me now.